utils/tests_fetcher.py

```diff
@@ -195,12 +195,22 @@ def get_test_dependencies(test_fname):
         content = f.read()
 
     # Tests only have relative imports for other test files
-    relative_imports = re.findall(r"from\s+\.(\S+)\s+import\s+([^\n]+)\n", content)
+    # TODO Sylvain: handle relative imports cleanly
+    relative_imports = re.findall(r"from\s+(\.\S+)\s+import\s+([^\n]+)\n", content)
     relative_imports = [test for test, imp in relative_imports if "# tests_ignore" not in imp]
-    dependencies = [os.path.join("tests", f"{test}.py") for test in relative_imports]
-    # Some tests use docstrings with relative imports in them and this could catch them, so we check the files
-    # actually exist
-    return [f for f in dependencies if os.path.isfile(f)]
+
+    # Removes the double trailing '..' for parent imports, and creates an absolute path from the root dir with
+    # `tests` as a prefix.
+    parent_imports = [imp.strip(".") for imp in relative_imports if ".." in imp]
+    parent_imports = [os.path.join("tests", f"{test.replace('.', '/')}.py") for test in parent_imports]
+
+    # Removes the single trailing '.' for current dir imports, and creates an absolute path from the root dir with
+    # tests/{module_name} as a prefix.
+    current_dir_imports = [imp.strip(".") for imp in relative_imports if ".." not in imp]
+    directory = re.findall(r"(.*)[/]", test_fname)[0]
```

Comment:
The / only works on Linux/Max, we need to use `os.path.sep` instead. How is this different from
```
directory = test_fname.split(os.path.sep)[0]
```
?

Reply:
The regex actually takes everything preceding the last forward slash. It is the equivalent of the following:
```py
directory = "/".join(test_fname.split("/")[:-1])
```
I have tested it by raising in case it wasn't true for all files in the repository:

```py
    directory = re.findall(r"(.*)[/]", test_fname)[0]
    if directory != "/".join(test_fname.split("/")[:-1]):
        raise ValueError(directory, test_fname(
```

As this is confirmed, I am replacing it by the non-regex as it is easier to understand what it does.

Reply:
What I need it to do is the following:

```
tests/tokenization/test_tokenization_fast.py -> tests/tokenization
tests/vit/test_feature_extraction_vit.py -> tests/vit
tests/test_modeling_common.py -> tests
```